README.md

```diff
@@ -1,289 +1,590 @@
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf#page=18).
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.  An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
 
-Specification for binding services to runtime applications running in Kubernetes.  
+## Terminology definition
 
-## Conventions
+<dl>
+  <dt>Duck Type</dt>
+  <dd>Any type that meets the contract defined in a specification, without being an instance of a specific concrete type.  For example, for specification that requires a given key on <code>status</code>, any resource that has that key on its <code>status</code> regardless of its <code>kind</code> would be considered a duck type of the specification.</dd>
 
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
-document are to be interpreted as described in [BCP
-14](https://tools.ietf.org/html/bcp14)
-[[RFC2119](https://tools.ietf.org/html/rfc2119)]
-[[RFC8174](https://tools.ietf.org/html/rfc8174)] when, and only when, they
-appear in all capitals, as shown here.
+  <dt>Service</dt>
+  <dd>Any software that exposes functionality.  Examples include an application with REST endpoints, an event stream, an Application Performance Monitor, or a Hardware Security Module.</dd>
 
-## Terminology definition
+  <dt>Application</dt>
+  <dd>Any process, running within a container.  Examples include a Spring Boot application, a NodeJS Express application, or a Ruby Rails application.  <b>Note:</b> This is different than an umbrella application as defined by the Kubernetes SIG, which refers to a set of micro-services.</dd>
+
+  <dt>Service Binding</dt>
+  <dd>The act of or representation of the action of providing information about a Service to an Application</dd>
+
+  <dt>ConfigMap</dt>
+  <dd>A Kubernetes <a href="https://kubernetes.io/docs/concepts/configuration/secret/">ConfigMap</a></dd>
+
+  <dt>Secret</dt>
+  <dd>A Kubernetes <a href="https://kubernetes.io/docs/concepts/configuration/configmap/">Secret</a></dd>
+</dl>
+
+# Provisioned Service
+
+A Provisioned Service resource **MUST** define a `.status.binding.name` which is a `LocalObjectReference` to a `Secret`.  The `Secret` **MUST** be in the same namespace as the resource.  The `Secret` **MUST** contain a `kind` entry with a value that identifies the abstract classification of the binding.  It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` **MAY** contain any other entry.
```

Comment:
Hi! Are we proposing that all service controllers/CRDs undergo code changes to adhere to this?

Or are you implying,
To be called a 'provisioned service', the corresponding resource  in the context of service binding must have the `.status.binding` section?

Reply:
> The `Secret` **MUST** contain a `kind` entry with a value that identifies the abstract classification of the binding. 

Since I'm catching up, could you help me with an example here?

Reply:
This proposal merged the metadata (ConfigMap) and secret (Secret) values into a single Secret resource. The metadata previously required `kind`, so now it is required on the secret.

The `kind` property indicates what kind of service the binding is for: mysql, postgres, mongodb, etc. It allows application to consume service by type, rather than by name.

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: production-db
stringData:
  kind: mysql
  username: root
  password: root
  hostname: localhost
  port: 3306
```

Reply:
+1

I like the merge of the metadata and the secret, as it leads to easier implementation from both the bindable service and the mounting side.

Reply:
At the most primitive level, for interoperability, types should have `.status.binding.name`.  How this entry in the resource definition comes to be, is *intentionally* undefined.  So if I was writing a green-field resource tomorrow, I might just add it and remove the need for any other participants in the interaction.

If I was working with an existing resource type or CRD, I'd probably start by *not* changing the resource.  Instead, I'd begin with the extension that talks about descriptors and annotations to dynamically create this value, without changing the resource/CRD (other than annotating it).  This gives a non-invasive on-ramp to bootstrap the ecosystem and possibly, in the long term, remove the need for annotations or other strategies like synthetic provisioned services.

Reply:
A couple of examples of Kind/Provider pairings:

* MySQL/ClearDB
* MySQL/Oracle
* MySQL/MicrosoftAzure
* PostgreSQL/ElephantSQL
* PostgreSQL/GoogleCloud
* PostgreSQL/AmazonRDS
* NewRelic
* Datadog
* LunaHSM/Gemalto
* ProtectApp/Gemalto
* StackdriverDebugger/Google
* AzureApplicationInsights/MicrosoftAzure
* JProfiler/ejTechnologies
* JaCoCo